src/components/LoginDialog.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <q-card style="min-width: 25%">
+    <q-card-section>
+      <div class="text-h6">
+        Log in
+      </div>
+    </q-card-section>
+    <form @submit.prevent="onSubmit">
+      <q-card-section>
+        <q-input
+          v-model="username"
+          dense
+          outlined
+          placeholder="Username"
+          stack-label="Stack Label"
+          :rules="[val => val.length<=15 || 'Usernames should not be over 15 characters']"
+        />
+      </q-card-section>
+      <q-card-section>
+        <q-input
+          v-model="password"
+          type="password"
+          dense
+          outlined
+          placeholder="Password"
+          stack-label="Stack Label"
+          :rules="[val => val.length<=15 || 'Passwords should not be over 15 characters']"
```

Comment:
passwords should probably be allowed longer. Maybe 20 to match usernames? As we're hashing them with Argon2, we may not need to limit password length

Reply:
Ah yes, I'll go with 20, to be honest I mostly added this limitation because I thought it looked bad that the user could input an unlimited number of characters without any counteraction, but let me know if you want me to do something different